azurerm/internal/services/policy/data_source_policy_set_definition.go

```diff
@@ -0,0 +1,141 @@
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+				ExactlyOneOf: []string{"name", "display_name"},
+			},
+
+			"name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+				ExactlyOneOf: []string{"name", "display_name"},
+			},
+
+			"management_group_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"metadata": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"parameters": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"policy_definitions": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"policy_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceArmPolicySetDefinitionRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Policy.SetDefinitionsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	displayName := d.Get("display_name").(string)
+	managementGroupID := d.Get("management_group_id").(string)
+
+	var setDefinition policy.SetDefinition
+	var err error
+
+	if displayName != "" {
+		setDefinition, err = getPolicySetDefinitionByDisplayName(ctx, client, displayName, managementGroupID)
+		if err != nil {
+			return fmt.Errorf("failed to read Policy Set Definition (Display Name %q): %+v", displayName, err)
+		}
+	}
+	if name != "" {
+		setDefinition, err = getPolicySetDefinition(ctx, client, name, managementGroupID)
+		if err != nil {
+			return fmt.Errorf("failed to read Policy Set Definition %q: %+v", name, err)
+		}
+	}
+
+	d.SetId(*setDefinition.ID)
+	d.Set("name", setDefinition.Name)
+	d.Set("display_name", setDefinition.DisplayName)
+	d.Set("description", setDefinition.Description)
+	d.Set("policy_type", setDefinition.PolicyType)
+	d.Set("metadata", flattenJSON(setDefinition.Metadata))
+	d.Set("parameters", flattenJSON(setDefinition.Parameters))
+
+	definitionBytes, err := json.Marshal(setDefinition.PolicyDefinitions)
+	if err != nil {
+		return fmt.Errorf("unable to flatten JSON for `policy_defintions`: %+v", err)
+	}
+	d.Set("policy_definitions", string(definitionBytes))
+
+	return nil
+}
+
+func getPolicySetDefinitionByDisplayName(ctx context.Context, client *policy.SetDefinitionsClient, displayName, managementGroupID string) (policy.SetDefinition, error) {
+	var setDefinitions policy.SetDefinitionListResultIterator
+	var err error
+
+	if managementGroupID != "" {
+		setDefinitions, err = client.ListByManagementGroupComplete(ctx, managementGroupID)
+	} else {
+		setDefinitions, err = client.ListComplete(ctx)
+	}
+	if err != nil {
+		return policy.SetDefinition{}, fmt.Errorf("failed to load Policy Definition List: %+v", err)
+	}
+
+	for setDefinitions.NotDone() {
```

Comment:
If we found more then 1 result could we return an error?

Reply:
Done